### ISSUE TITLE
MIDIOut: Index Out of Range error

### ISSUE THREAD

**Issue by akstuhl** [supercollider/supercollider#4652]
The `_SendMIDIOut` primitive throws an Index Out of Range error when addressing an output port whose index is equal to or greater than the number of input ports. I believe this is because the method checks `outputIndex >= gNumMIDIInPorts` when it instead should check `gNumMIDIOutPorts`: https://github.com/supercollider/supercollider/blob/f806ace7bd8565dd174e7d47a1b32aaa4175a46e/lang/LangPrimSource/SC_CoreMIDI.cpp#L765

## Environment

* SuperCollider version: 3.10.3
* Operating system: Mac OS 10.14.6
* Other details (Qt version, audio driver, etc.): Two MIDI devices connected (BCR2000 and Launchpad). The BCR2000 has 2 input ports and 3 output ports. The Launchpad appears at the end of the MIDI destination lists.

## Steps to reproduce

```supercollider
~midiout = MIDIOut.newByName("Launchpad S", "Launchpad S");
~midiout.touch;
```

## Expected vs. actual behavior

Expected: touch MIDI device.
Actual: `ERROR: Primitive '_SendMIDIOut' failed. Index out of range.`

Removing the first MIDI device (the BCR2000) and re-initializing MIDI allows the same code to be run without error.

**Comment by dmartinp** [comment]
I have run into this same error... Thanks for tracking this down.

**Comment by brianlheim** [comment]
thanks for the report @akstuhl -- your solution seems right to me, please file a PR if you have the time!

**Comment by brianlheim** [comment]
fixed in #4910.
